wolfcrypt/test/test.c

```diff
@@ -8009,7 +8009,7 @@ static int aes_cbc_test(void)
     byte cipher[AES_BLOCK_SIZE];
     byte plain[AES_BLOCK_SIZE];
     int  ret;
-    static const byte msg[] = { /* "Now is the time for all " w/o trailing 0 */
+    const byte msg[] = { /* "Now is the time for all " w/o trailing 0 */
```

Comment:
@douzzer , didn't you recently change these to "static" and we had a discussion about it? @tmael is having a run-time issue with static because the memory comes from a different region. Can you remind us why "static" is needed here (I think LKM tests)? If so then we should come up with a build macro to optionally place it.

Reply:
@dgarske the idea is that if a byte vector is declared mutable, then it will go on the stack in its entirety, and the initialization value will get copied from the object's data segment onto the stack, whereas if it's `static`, the compiler can just use the initializer in the data segment directly, with no stack footprint.

Given it's `const`, obviously there really are no write ops to it.  Some compilers (read: gcc & clang) may notice this and realize the vector doesn't need to get duped onto the stack, but that would only be a risk-free optimization on targets with a TLB (virtual memory).

Seems to me that my manual `static` optimizations are only needed in the `WOLFSSL_SMALL_STACK` scenario, and it sounds like additional gating may be needed.  I'd be inclined to gate any of these that are or might be causing problems with a new gate, `WOLFSSL_SMALL_STACK_STATIC` or somesuch.